x-pack/agent/pkg/core/plugin/process/process.go

```diff
@@ -188,5 +195,9 @@ func pushCredentials(w io.Writer, c *Creds) error {
 	}
 
 	_, err = w.Write(credbytes)
+
+	// this gives beat with grpc a bit of time to spin up a goroutine and start a server.
+	// should be ok until we come up with more clever solution.
```

Comment:
Lets create an issue with the idea of adding a status endpoint as part of our grpc.